Modules/MediaPool/classes/Upload/class.ilMediaPoolBulkUploadGUI.php

```diff
@@ -0,0 +1,230 @@
+                    $mob = new ilObjMediaObject();
+                    $mob->setTitle($title);
+                    $mob->setDescription("");
+                    $mob->create();
+                    
+                    $mob->createDirectory();
+                    $media_item = new ilMediaItem();
+                    $mob->addMediaItem($media_item);
+                    $media_item->setPurpose("Standard");
+                    
+                    $mob_dir = ilObjMediaObject::_getRelativeDirectory($mob->getId());
+                    $file_name = ilObjMediaObject::fixFilename($title);
+                    $file = $mob_dir . "/" . $file_name;
+                    
+                    $this->upload->moveOneFileTo(
+                        $result,
+                        $mob_dir,
+                        Location::WEB,
+                        $file_name,
+                        true
+                    );
+                    
+                    $mep_item = new ilMediaPoolItem();
+                    $mep_item->setTitle($title);
+                    $mep_item->setType("mob");
+                    $mep_item->setForeignId($mob->getId());
+                    $mep_item->create();
+                    
+                    $tree = $this->media_pool->getTree();
+                    $parent = $this->mep_item_id;
+                    $tree->insertNode($mep_item->getId(), $parent);
+                    
+                    // get mime type
+                    $format = ilObjMediaObject::getMimeType($file);
+                    $location = $file_name;
+                    
+                    // set real meta and object data
+                    $media_item->setFormat($format);
+                    $media_item->setLocation($location);
+                    $media_item->setLocationType("LocalFile");
+                    $media_item->setUploadHash($this->upload_hash); // TODO CHECK
+                    $mob->update();
+                    
+                    $item_ids[] = $mob->getId();
+                    
+                    $mob = new ilObjMediaObject($mob->getId());
+                    $mob->generatePreviewPic(320, 240);
+                    
+                    // duration
+                    $med_item = $mob->getMediaItem("Standard");
+                    $med_item->determineDuration();
+                    $med_item->update();
+                    
+                    $result = new BasicHandlerResult(
+                        $this->getFileIdentifierParameterName(),
+                        BasicHandlerResult::STATUS_OK,
+                        $med_item->getId(),
+                        ''
+                    );
+                }
+            } catch (Exception $e) {
+                $result = new BasicHandlerResult(
+                    $this->getFileIdentifierParameterName(),
+                    BasicHandlerResult::STATUS_FAILED,
+                    '',
+                    $e->getMessage()
+                );
```

Comment:
Also this BasicHandlerResult and whole logic that needs to be implemented to handle uploads. Is this part of any README file? I did not find anything.

Reply:
see https://github.com/ILIAS-eLearning/ILIAS/blob/trunk/Services/FileServices/README.md

Reply:
Thanks.